File: chinese-shadowing-client/src/models/Room.js
import Shadow from './Shadow';
import UserPersona from './UserPersona';

/** @enum {string} */
export const RoomState = {
  ACTIVE: 'Active',
  COMPLETED: 'Completed',
  CANCELLED: 'Cancelled',
};

export default class Room {
  /** @type {string} */
  id;
  /** @type {string} */
  episodeId;
  /** @type {UserPersona[]} */
  userPersonas;
  /** @type {Shadow[]} */
  shadows;
  /** @type {RoomState} */
  state;

  constructor(room) {
    this.id = room.id;
    this.episodeId = room.episodeId;
    this.userPersonas = room.userPersonas.map(x => new UserPersona(x));
    this.shadows = room.shadows.map(x => new Shadow(x));
    this.state = room.state;
  }
}

File: chinese-shadowing-api/Shadowing.Models/Rooms/ViewModels/Room.cs
using Shadowing.Models.Rooms.Enums;
using Shadowing.Models.Shadows.ViewModels;
using Shadowing.Models.UserPersonas.ViewModels;
using System.Collections.Generic;

namespace Shadowing.Models.Rooms.ViewModels
{
    public class Room
    {
        public string Id { get; set; }
        public string EpisodeId { get; set; }
        public List<UserPersona> UserPersonas { get; set; }
        public List<Shadow> Shadows { get; set; }
        public RoomState State { get; set; }
        public string CreatedById { get; set; }
    }
}
